ampd/src/sui/verifier.rs

```diff
@@ -18,11 +16,33 @@ struct ContractCall {
     pub payload_hash: Hash,
 }
 
-// Event type is in the form of: <gateway_address>::gateway::ContractCall
-fn call_contract_type(gateway_address: &SuiAddress) -> StructTag {
-    format!("{}::{}", gateway_address, CONTRACT_CALL_EVENT)
-        .parse()
-        .expect("failed to parse struct tag")
+#[derive(Deserialize)]
+struct OperatorshipTransferred {
+    pub payload: Vec<u8>,
```

Comment:
Not related to this PR, but why does the Sui contract's event look like this? If the `OperatorshipTransferred` is done this way, why isn't `ContractCall` done the same? It could have just been a blob of json for literally anything.

Reply:
yes, it a blob of bytes of contains new pub keys and threshold. idk, can definately be more structured.